### ISSUE TITLE
thymeleaf viewresolver is ignored by spring webflow

### ISSUE THREAD

**Issue by MaggiWuerze** [thymeleaf/thymeleaf-spring#204]
Spring: 4.3.18.RELEASE
Webflow: 2.5.1.RELEASE
Thymeleaf: 3.0.9.RELEASE

I am currently setting up an web Application with Spring4 + Thymeleaf and now i want to add Spring Webflow functionality using html pages.

The flows get triggerd when i call their URL but i always get the following error message:

    `java.lang.IllegalArgumentException: Unsupported view type /WEB-INF/templates/test.html only types supported by this FlowViewResolver implementation are [.jsp] and [.jspx]`

I added the java config to use a custom viewresolver as described in the Thymeleaf [Documentation](https://www.thymeleaf.org/doc/tutorials/3.0/thymeleafspring.html#spring-webflow-integration):

```
@Bean
    public MvcViewFactoryCreator mvcViewFactoryCreator() {
        MvcViewFactoryCreator factoryCreator = new MvcViewFactoryCreator();

        factoryCreator.setViewResolvers(
                Collections.singletonList(thymeleafViewResolver));

        factoryCreator.setUseSpringBeanBinding(true);
        return factoryCreator;
    }
```
```
@Bean
    public ViewResolver thymeleafViewResolver() {
        AjaxThymeleafViewResolver resolver = new AjaxThymeleafViewResolver();
        resolver.setTemplateEngine(templateEngine());
        resolver.setViewClass(FlowAjaxThymeleafView.class);
        resolver.setCharacterEncoding(UTF8);
        resolver.setOrder(0);
        return resolver;
    }
```

When I debug the call i find that instead of the viewResolver from the config, Webflow instead uses the default FlowResourceFlowViewResolver.

EDIT: updated the viewResolver Bean to include (didn't help)
  
        resolver.setViewClass(FlowAjaxThymeleafView.class);

**Comment by danielfernandez** [comment]
After the latest changes in Thymeleaf `3.0.12-SNAPSHOT` for #208, integration works for me in a Spring Boot 2.4.0 application with Spring WebFlow 2.5.0.

But documentation is quite outdated and incomplete as there are a number of beans involved in Spring WebFlow configuration that are not specified in the Thymeleaf docs. We will consider this a documentation issue therefore.

**Comment by danielfernandez** [comment]
Closed by means of thymeleaf/thymeleaf-docs@71f137f.

After some testing, the only way that an error such as yours can happen is if `org.springframework.webflow.mvc.builder.FlowResourceFlowViewResolver` is used instead of an instance of `org.springframework.webflow.mvc.builder.DelegatingFlowViewResolver` that delegates to an `AjaxThymeleafViewResolver`. But this should never happen because this call you have in your code actually creates such _delegate_ instance wrapping the Thymeleaf view resolver:

```java
        factoryCreator.setViewResolvers(
                Collections.singletonList(thymeleafViewResolver));
```

But what's stranger, your appplication should never have even reached that point because Thymeleaf 3.0.9 is not compatible with Spring WebFlow 2.5 as mentioned in #208 and you should have received a `ClassNotFoundException` on application startup. So I believe your issue was somewhere else.

I've improved the documentation section on Spring WebFlow configuration in order to: **1.** State the fact that now Thymeleaf will require Spring WebFlow 2.5+; and **2.** Better explain how all the thymeleaf-related beans must be created.

Requiring Spring WebFlow 2.5+ seems like a safe option taking into account that Spring WebFlow 2.5 was released more than 2 years ago.
